### PR TITLE
refactor: write integration tests in Golang

### PR DESCRIPTION
**Pull request by tisonkun**
## Motivation

TCL scripts are hard for debugging. It helps to bring Redis tests but we can later refactor them for easy maintaining and debugging.

I ever try to write it in Rust https://github.com/apache/incubator-kvrocks/discussions/693 but finally propose this patch in Golang. As Rust's testing is less than mature and our contributors like @git-hulk & @torwig should be familiar with Golang. Also, go-redis project has [a backlink to Kvrocks](https://github.com/go-redis/redis#ecosystem) and when possibly, I'm glad to use it for Kvrocks' tests.

## Implementation

1. Add go integration cases mod under `tests/gocase`.
2. Implement the start server logic at `util/server.go`
3. Migrate TCL test `util/command` to the new integration framework.
4. Wire logic to the CI workflow.

### CODE REVIEW

**Comment by git-hulk** [comment]
Thanks @tisonkun

The tcl is really hard to debug for myself. Go or Rust are good to me but I think Go will be easier for developers to learn and participate. Can see other guys have any feedback on this topic,

File: .github/workflows/kvrocks.yaml
Comment (PragmaTwice) [inline .github/workflows/kvrocks.yaml:174]: Could these be added in `./x.py`, like `./x.py test go <build-dir>`?

Comment (tisonkun) [reply .github/workflows/kvrocks.yaml:174]: Yes, it's possible. I'd like to do it as a follow-up, though.

Comment (PragmaTwice) [reply .github/workflows/kvrocks.yaml:174]: got it

File: tests/tcl/tests/test_helper.tcl
Comment (PragmaTwice) [inline tests/tcl/tests/test_helper.tcl:59]: If we delete it in the migration procedure, developers may need a two phase testing in their local:
1. run tcl tests
2. run go tests

Any idea to simplify the test executing?

Comment (tisonkun) [reply tests/tcl/tests/test_helper.tcl:59]: If we don't delete it, the situation doesn't change. If only the migration can be eventually closed, I don't think it's a big deal.

Since we do have TCL tests and go IT cases, we always run both. There's no way to avoid it. Only you may batch them in one script call but it seems not necessary.

Comment (PragmaTwice) [reply tests/tcl/tests/test_helper.tcl:59]: Sounds good to me.

File: tests/gocase/util/server.go
Comment (PragmaTwice) [inline tests/gocase/util/server.go]: I think `kvrocks.conf` is enough rather than a random string? We can locate it more easy if it has a fixed name.

Comment (tisonkun) [reply tests/gocase/util/server.go]: Good point! Will push a commit for it.

Comment (tisonkun) [reply tests/gocase/util/server.go]: Updated.

**Comment by git-hulk** [comment]
I looked through this PR and it’s clear enough for me. To see other guys have any questions? @PragmaTwice @ShooterIT @Alfejik @caipengbo

**Review by PragmaTwice — APPROVED** [review]
LGTM

**Comment by tisonkun** [comment]
It happens that I have some time to work on this migration. I'll merge this patch tomorrow if no more objections and continue the tasks.

Feel free to review after merge.

**Review by caipengbo — APPROVED** [review]

**Pull request merged**
